polars_st/geodataframe.py

```diff
@@ -339,12 +339,24 @@ def to_geopandas(
         """Convert this DataFrame to a geopandas GeoDataFrame."""
         import geopandas as gpd
 
+        if self._df.is_empty():
+            crs = None
+        else:
+            try:
+                crs = self._df.select(geom(geometry_name).st.srid()).unique().item()
+            except ValueError:
+                # Multiple SRIDs found
+                msg = "DataFrame with mixed SRIDs aren't supported in GeoPandas"
+                raise ValueError(msg)
```

Comment:
Thanks for your PRs and apologies about the nitpicks. The empty frame case doesn't need to be handled separately if we only use the amount of unique srids. Also, I don't like relying on exception handling for index access when it's just as easy to validate the series length beforehand. Finally, `unique` will include nulls, so a dataframe with uniform SRIDs but null rows will fail this test.

```suggestion

        srids = self._df.select(geom(geometry_name).st.srid()).unique().drop_nulls().to_series()
        match srids.len():
            case 0:
                crs = None
            case 1:
                crs = srids.item()
            case _:
                msg = "DataFrame with mixed SRIDs aren't supported in GeoPandas"
                raise ValueError(msg)
```